### PR TITLE
feat(spooler): Improve docs for the spool configuration

### PR DESCRIPTION
**Pull request by iambriccardo**
#skip-changelog

### CODE REVIEW

File: relay-server/src/services/buffer/envelope_buffer/mod.rs [hunk 1]
```diff
@@ -299,7 +299,7 @@ where
     /// If the envelope stack does not exist, a new stack is pushed to the priority queue.
     /// The priority of the stack is updated with the envelope's received_at time.
     pub async fn push(&mut self, envelope: Box<Envelope>) -> Result<(), EnvelopeBufferError> {
-        let received_at = envelope.meta().received_at();
```
Comment (iambriccardo) [inline relay-server/src/services/buffer/envelope_buffer/mod.rs:302]: Drive-by refactor.

File: relay-config/src/config.rs [hunk 2]
```diff
@@ -958,7 +967,6 @@ impl Default for EnvelopeSpool {
         Self {
             path: None,
             max_disk_size: spool_envelopes_max_disk_size(),
-            max_memory_size: spool_envelopes_max_memory_size(),
```
Comment (iambriccardo) [inline relay-config/src/config.rs:961]: Not needed anymore.

File: relay-config/src/config.rs [hunk 3]
```diff
@@ -905,41 +900,50 @@ fn spool_envelopes_partitions() -> NonZeroU8 {
...
     ///
-    /// If set, this will enable the buffering for incoming envelopes.
+    /// Based on the number of partitions, more database files will be created within the same path.
+    ///
+    /// If not set, the envelopes will be buffered in memory.
     pub path: Option<PathBuf>,
     /// The maximum size of the buffer to keep, in bytes.
     ///
-    /// If not set the default is 524288000 bytes (500MB).
+    /// When the buffer, set to use SQLite, reaches this disk capacity, new envelopes will be dropped.
```
Comment (loewenheim) [inline relay-config/src/config.rs]: Nit:
```suggestion
    /// When the on-disk buffer reaches this size, new envelopes will be dropped.
```

**Review by loewenheim — APPROVED** [review]
